src/Features/CSharp/Portable/Completion/SuggestionMode/CSharpSuggestionModeCompletionProvider.cs

```diff
@@ -110,6 +110,12 @@ private bool IsAnonymousObjectCreation(SyntaxToken token)
 
         private bool IsLambdaExpression(SemanticModel semanticModel, int position, SyntaxToken token, ITypeInferenceService typeInferrer, CancellationToken cancellationToken)
         {
+            // Not after `new`
+            if (token.IsKind(SyntaxKind.NewKeyword) && token.Parent.IsKind(SyntaxKind.ObjectCreationExpression))
```

Comment:
i don't think you have to ever check the parent.  it's simply never a lambda expression context after 'new'.